pennylane/__init__.py

```diff
@@ -340,3 +307,45 @@ def run_cnot():
 def version():
     """Returns the PennyLane version number."""
     return __version__
+
+
+# add everything as long as it's not a module and not prefixed with _
+_all = sorted(
+    [
+        name
+        for name, function in globals().items()
+        if not (name.startswith("_") or isinstance(function, types.ModuleType))
+    ]
+)
+
+
+_qchem = None
+
+
+def __getattr__(name):
```

Comment:
I performed VQE calculations in the following cases, everything seems to work as expected.
A. PennyLane-QChem is installed,
    i. `from pennylane import qchem` works fine
    ii. `qml.qchem.excitations(2, 4)` without importing qchem works fine
B. PennyLane-QChem is `not` installed
    i. `from pennylane import qchem`: does not complain
    ii. `qml.qchem.excitations(2, 4)` with/without importing qchem: asks for installing qchem